snuba/snuba_migrations/spans/0001_spans_v1.py

```diff
@@ -0,0 +1,99 @@
+from typing import List, Sequence
+
+from snuba.clickhouse.columns import UUID, Column, DateTime, Nested, String, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from snuba.migrations.operations import OperationTarget, SqlOperation
+from snuba.utils.schemas import Float
+
+storage_set_name = StorageSetKey.SPANS
+local_table_name = "spans_local"
+dist_table_name = "spans_dist"
+
+UNKNOWN_SPAN_STATUS = 2
+
+columns: List[Column[Modifiers]] = [
+    Column("project_id", UInt(64)),
+    Column("transaction_id", UUID(Modifiers(nullable=True))),
+    Column("trace_id", UUID()),
+    Column("span_id", UInt(64)),
+    Column("parent_span_id", UInt(64, Modifiers(nullable=True))),
+    Column("segment_id", UInt(64)),
+    Column("is_segment", UInt(8)),
```

Comment:
I'm a little confused as to why this is needed. In this model are spans and segments the same entity demarcated by this flag?

Reply:
Yes. segments are the root spans from which a trace starts. It is not part of the list of spans sent from upstream. Look at the example trace in the description to see what a segment is.